memgpt/local_llm/chat_completion_proxy.py

```diff
@@ -67,17 +71,20 @@ def get_chat_completion(
             # make the default to use grammar
             llm_wrapper = DEFAULT_WRAPPER(include_opening_brace_in_prefix=False)
             # grammar_name = "json"
-            grammar_name = "json_func_calls_with_inner_thoughts"
+
+            grammar, documentation = generate_grammar_and_documentation(functions_python)
+            printd(grammar)
         else:
             llm_wrapper = DEFAULT_WRAPPER()
     elif wrapper not in available_wrappers:
         raise ValueError(f"Could not find requested wrapper '{wrapper} in available wrappers list:\n{available_wrappers}")
     else:
         llm_wrapper = available_wrappers[wrapper]
-        if "grammar" in wrapper:
-            grammar_name = "json_func_calls_with_inner_thoughts"
-
-    if grammar_name is not None and endpoint_type not in ["koboldcpp", "llamacpp", "webui"]:
+        if endpoint_type in ["koboldcpp", "llamacpp", "webui"]:

```

Comment:
For the meantime can we change this back to 
```python
if "grammar" in wrapper:
    ...
```

or
```python
if "grammar" in wrapper:
    if endpoint_type not in ["koboldcpp", "llamacpp", "webui"]:
        print(f"{CLI_WARNING_PREFIX}grammars are currently not supported when using {endpoint_type} as the MemGPT local LLM backend")
    else:
        ...
```

So that the user has to enable the grammar intentionally (instead of it being enabled by default)?

I think in a follow-up PR, we can pull grammar out into its own LLM config field or something, so that the user can enable grammars by running with `memgpt run --grammar`.